docs/design.md

```diff
@@ -221,6 +221,50 @@ NOTE: The `saved-model/` directory is the same as the container for the Model ob
 /content/saved-model/       # Mounted RO: from .spec.model
 ```
 
+### Naming
+
+Jobs, Deployments, and Pods have a 63 character .metadata.name limit.
+Datasets, Models, Servers, and Notebooks create these resources.
+It is nice to name child resources something that easily identifies them
+as belonging to a parent resource (i.e. similar name).
+In order to do this, we need to limit the length of the parent resource name.
+Suffixes should be short:
+
+```
+{parent.name}-{suffix}
```

Comment:
This had me think maybe tracking a `.status.parent` is worthwhile on models and datasets. Captured initial thoughts here: https://github.com/substratusai/substratus/issues/151